infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -497,7 +497,7 @@ private synchronized void startSnapshotApply(LogReplicationEntryMsg entry) {
 
         // Sync with registry after transfer phase to capture local updates, as transfer phase could
         // take a relatively long time.
-        replicationContext.refresh();
+        replicationContext.refreshConfig(session, false);
```

Comment:
Earlier, there was a check which regenerated the config only if the registry table has new entries.  I may be missing something but did not find it in ConfigManager in this PR.  

I suppose that the extra argument for session was added so that config for only a given session is refreshed.  But regenerating the config for all sessions can have an advantage because one refresh() call takes care of all sessions.  If there are not new entries in the registry table, refresh() will be a no-op subsequently.

Reply:
yes that part is changed in this PR. I considered several options:
1. Previous implementation, check if there are new entries and if so, update all sessions' config. It's no longer suitable now because we are also checking group-destinations table, and the entries in this table have more side effects (required snapshot sync for impacted sessions). So I think it's better to let the Session itself to find out it's impacted and only update its own config (instead of updating all session's config)
2. Current implementation, there is still a check for whether or not to read RegistryTable/ClientConfigTable and update in-memory entries (these 2 methods are synchronized and every session could compete on it) - but there is no check for using in-memory entries to update the config, because it could be the case that one session updates the in-memory entries, but other sessions haven't updated their config with latest entries.
3. Based on method 2, maybe we can add a session-logtail map if we still want to update config only when it's necessary

I thought option 2 should be fine since the in-memory calculation could be trivial, but open to ideas.